spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -835,6 +842,186 @@ public void setSchemaType(SchemaType schemaType) {
+			this.authPluginClassName = authPluginClassName;
+		}
+
+		public String getAuthParams() {
+			return this.authParams;
+		}
+
+		public void setAuthParams(String authParams) {
+			this.authParams = authParams;
+		}
+
+		public String getTlsTrustCertsFilePath() {
+			return this.tlsTrustCertsFilePath;
+		}
+
+		public void setTlsTrustCertsFilePath(String tlsTrustCertsFilePath) {
+			this.tlsTrustCertsFilePath = tlsTrustCertsFilePath;
+		}
+
+		public boolean isTlsAllowInsecureConnection() {
+			return this.tlsAllowInsecureConnection;
+		}
+
+		public void setTlsAllowInsecureConnection(boolean tlsAllowInsecureConnection) {
+			this.tlsAllowInsecureConnection = tlsAllowInsecureConnection;
+		}
+
+		public boolean isTlsHostnameVerificationEnable() {
+			return this.tlsHostnameVerificationEnable;
+		}
+
+		public void setTlsHostnameVerificationEnable(boolean tlsHostnameVerificationEnable) {
+			this.tlsHostnameVerificationEnable = tlsHostnameVerificationEnable;
+		}
+
+		public boolean isUseKeyStoreTls() {
+			return this.useKeyStoreTls;
+		}
+
+		public void setUseKeyStoreTls(boolean useKeyStoreTls) {
+			this.useKeyStoreTls = useKeyStoreTls;
+		}
+
+		public String getSslProvider() {
+			return this.sslProvider;
+		}
+
+		public void setSslProvider(String sslProvider) {
+			this.sslProvider = sslProvider;
+		}
+
+		public String getTlsTrustStoreType() {
+			return this.tlsTrustStoreType;
+		}
+
+		public void setTlsTrustStoreType(String tlsTrustStoreType) {
+			this.tlsTrustStoreType = tlsTrustStoreType;
+		}
+
+		public String getTlsTrustStorePath() {
+			return this.tlsTrustStorePath;
+		}
+
+		public void setTlsTrustStorePath(String tlsTrustStorePath) {
+			this.tlsTrustStorePath = tlsTrustStorePath;
+		}
+
+		public String getTlsTrustStorePassword() {
+			return this.tlsTrustStorePassword;
+		}
+
+		public void setTlsTrustStorePassword(String tlsTrustStorePassword) {
+			this.tlsTrustStorePassword = tlsTrustStorePassword;
+		}
+
+		public String[] getTlsCiphers() {
+			return this.tlsCiphers;
+		}
+
+		public void setTlsCiphers(String[] tlsCiphers) {
+			this.tlsCiphers = tlsCiphers;
+		}
+
+		public String[] getTlsProtocols() {
+			return this.tlsProtocols;
+		}
+
+		public void setTlsProtocols(String[] tlsProtocols) {
+			this.tlsProtocols = tlsProtocols;
+		}
+
+		public long getConnectionTimeoutMs() {
+			return this.connectionTimeoutMs;
+		}
+
+		public void setConnectionTimeoutMs(long connectionTimeoutMs) {
+			this.connectionTimeoutMs = connectionTimeoutMs;
+		}
+
+		public long getRequestTimeoutMs() {
+			return this.requestTimeoutMs;
+		}
+
+		public void setRequestTimeoutMs(long requestTimeoutMs) {
+			this.requestTimeoutMs = requestTimeoutMs;
+		}
+
+		public Map<String, Object> buildProperties() {
+			PulsarProperties.Properties properties = new Properties();
+
+			PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
+
+			map.from(this::getServiceUrl).to(properties.in("serviceUrl"));
```

Comment:
So the admin client and regular client both take in the `org.apache.pulsar.client.impl.conf.ClientConfigurationData` and have almost identical props here in our configuration of them. 

**Question 1**: Should they support the same exact properties? 
If so, we could consolidate some of the code w/ a common client props base class.

**Question 2**: Should each one support all of the properties in the clientconfigdata? 
if not, what is the final arbiter on which client support which options? I would think that the pulsar AdminBuilder / ClientBuilder provides that contract w/ its builder methods. 

If the builders are the contract, then one thought would be to populate their builders rather than the raw map. 


Whatever we decide here, we should circle back through the regular client properties @sobychacko and make sure we have them all covered. I do see some missing ones at 1st glance: 
```java
numListenerThreads
maxLookupRedirects
authParamMap
socks5ProxyAddress/Username/Password
```

Reply:
Me and @sobychacko discussed offline and came to the conclusion to make the `ClientConfigurationData` the source of truth for the supported properties for both admin and regular pulsar client. 

Let's also go w/ this set of mapped properties for the initial offering. We will create a follow-on issue to revisit and fine-tune the properties to be sure we are exposing the ones we want for both admin and regular client.

@alpreu please resolve this comment once you have digested it.

Reply:
Following are some thoughts I have on this. The `ClientConfigurationData` being used for both the admin as well as the regular client is a misuse on our side. 
First, not all properties are shared between both clients, for example, the SOCKS5 proxy properties are only used on the regular client. 
Further, there are properties that even though they are shared, behave differently for the clients. An example of this is the `serviceUrl`, which for the regular client will look like `pulsar+ssl://localhost:6650`, while for the admin it will look like `https://localhost:8080`.
This makes re-using the same set of properties for both difficult.

Reply:
By `source of truth for the supported properties for both admin and regular pulsar client` I really just mean these are the properties we would support. Thanks for the info on their divergence (admin and regular client) @alpreu .

In this case, I bet the actual AdminBuilder and PulsarClientBuilder are the actual final arbiter of what is supported. It feels to me like we should support whatever their respective builders support. 

For this PR I think what we have is fine for 1st pass. We should create a follow on ticket to explore this a bit more.

Reply:
Will address in #76

Reply:
Yes, it is a good idea to go through the properties again. 
I also filed a bug ticket in Pulsar for the properties that get overridden by the builder's default values.

Reply:
> Yes, it is a good idea to go through the properties again. I also filed a bug ticket in Pulsar for the properties that get overridden by the builder's default values.

Thank you for also filing the bug ticket in Pulsar.